rpc/export.go

```diff
@@ -95,26 +122,28 @@ func (c *Conn) sendCap(d rpccp.CapDescriptor, client *capnp.Client, state capnp.
 	// TODO(someday): Check for pipeline client on question for receiverAnswer.
 
 	// Default to sender-hosted (export).
-	for id, ent := range c.exports {
-		if ent == nil {
-			continue
-		}
-		if ent.client.IsSame(client) {
-			ent.wireRefs++
-			d.SetSenderHosted(uint32(id))
-			return exportID(id), true
-		}
+	state.Metadata.Lock()
+	defer state.Metadata.Unlock()
+	id, ok := c.findExportID(state.Metadata)
+	if ok {
+		ent := c.exports[id]
+		ent.wireRefs++
+		d.SetSenderHosted(uint32(id))
+		return id, true
 	}
+
+	// Not already present; allocate an export id for it:
 	ee := &expent{
 		client:   client.AddRef(),
 		wireRefs: 1,
 	}
-	id := exportID(c.exportID.next())
+	id = exportID(c.exportID.next())
```

Comment:
Shouldn't `next()` just return a correctly-typed `exportID`?  Again, visual noise.

Reply:
It was like that when I found it :P. But I think the reason is that `c.exportID` is of type `idgen`, which is used for generating other types of identifiers. Once generics lands, we could consider making idgen generic and changing this to be an `idgen(exportID)`. For now though, I think the code re-use outweights needing to cast.